src/infoManagers/endpointManager.ts

```diff
@@ -37,9 +37,11 @@ export class EndpointManager extends Disposable {
 	 * @param location the file location to encode.
 	 * @returns the encoded endpoint.
 	 */
-	public async encodeLooseFileEndpoint(location: string): Promise<string> {
+	public async encodeLooseFileEndpoint(location: string, file?: vscode.Uri): Promise<string> {
 		let fullParent = await PathUtil.GetParentDir(location);
-		const child = await PathUtil.GetFileName(location, true);
+		const child = file?.scheme === 'vscode-chat-code-block' ? 

```

Comment:
The code block URIs have a path like `/response_1/0`. `GetFileName` would just return `0` in that case and we should keep more context.

This is a hack but the path handling for non-file URIs is a little confusing. Maybe we can come up with something better.

Reply:
Technically this still isn't enough- the full URI is like `"vscode-chat-code-block://1a22dd3d-ff95-4b4b-bb77-ca04942e2280/response_0/0"` and the GUI identifies the chat session. The GUID should be part of the url too, otherwise you could have collisions between codeblocks in multiple chat sessions.

Reply:
would there be multiple files that return `0`? I guess it should be fine as long as they don't need the old preview, but we want to prevent server path duplications. IIRC, this is generating the path that the server uses to access the file.

Reply:
also, I'm a bit confused: if you used `file.fsPath.slice(1)` and the path was `/response_1/0`, wouldn't the string return `response_1/0`? Or is what you're trying to say is that you want to retain more than just `0`?

Reply:
Oh okay, I re-read and it does seem that you want to retain up to `response_1/0`. I don't mind the GUID being part of the URL too. Is that part of the authority?

Reply:
Yeah, the guid is part of the authority.

One way to think about it would be that we could have a nice path for Untitled files where it produces a readable url, but for other files that are not Untitled, not file: scheme, we just put the whole URI into the generated path, and it wouldn't be pretty but then it will work for all other types of URIs. 

Are there other types of URIs that you can preview that aren't Untitled editors but have some other scheme? Like from custom FS providers? I wasn't sure

Reply:
I don't think that we support any other types of schemes that require special treatment, so this would be a first. I'm fine with this handling for now.